qemu/tests/uefishell.py

```diff
@@ -165,6 +169,7 @@ def clean(self):
         To clean the test environment, including restore VAR file
         and close the session
         """
+        self.var_copy()
```

Comment:
Would you help illustrate why move this to `clean()`? Cause I remember that the developer has emphasized every single test case need a clean var file, so here I copy a new one at the beginning of the test case, just as what I did during manual test.

Reply:
Yes,but now we need to get the `file_ovmf_vars` from `vm.devices`, so we need to copy it after the guest is started and before the end of the test.Thanks!

Reply:
@leidwang If you mean you can not get the `file_ovmf_vars` before vm boot up, I suppose actually you can. this test case is relatively special, and from my perspective it is not a good choice to change the business logic only for fit certain test code design. Business logic is always in the first priority.
How to get it before vm boot up? You might 1. Get if from `params[params[''firmware_name'] + '_vars_filename']` and define different params['firmware_name'] for different architecture; Or 2. Utilize qcontainer.DevContainer directly without the vm object, but this requires you move `pflash_handler()` out of `machine_by_params()` first, so that you can use it to generate proper device, but I think that's not that necessary, approach 1 is enough for the requirement of this test case.

Reply:
@vivianQizhu @leidwang , you can use `restore_ovmf_vars = yes` to copy OVMF vars before the test starts.

Reply:
@PaulYuuu Yes, thanks for reminder, I don't remember if the order is fine to copy it during vm command line generating process, it should be okay, @leidwang you might have a try.

Reply:
@PaulYuuu @vivianQizhu Thank you all,will have a try.

Reply:
@vivianQizhu @PaulYuuu I think we can delete `var_copy()` directly and add `restore_ovmf_vars = yes` to cfg. Do you agree?Thanks a lot!

Reply:
@leidwang Yes, it's okay for me.